internal/pkg/plugin/gitlabci/java/options.go

```diff
@@ -0,0 +1,150 @@
+package java
+
+import (
+	"fmt"
+	"html/template"
+
+	"github.com/spf13/viper"
+)
+
+type RegistryType string
+
+const (
+	RegistryDockerhub RegistryType = "dockerhub"
+	RegistryHarbor    RegistryType = "harbor"
+)
+
+// Options is the struct for configurations of the gitlabci-java plugin.
+type Options struct {
+	PathWithNamespace string   `validate:"required"`
+	Branch            string   `validate:"required"`
+	BaseURL           string   `validate:"omitempty,url"`
+	Package           *Package `validate:"required"`
+	Build             *Build   `validate:"required"`
+	Deploy            *Deploy  `validate:"required"`
+}
+
+type BaseOption struct {
+	Enable        bool
+	Image         string
+	Tags          string
+	AllowedBranch []string
+}
+
+type Package struct {
+	*BaseOption
+	ScriptCommand []string
+}
+
+type Build struct {
+	*BaseOption
+	Registry      *Registry
+	ImageName     string
+	ScriptCommand []string
+}
+
+type Registry struct {
+	Type     RegistryType
+	Username string
```

Comment:
code: `Username`
doc: `userName`

This is inconsistent, and a better approach would be to explicitly specify the `mapstructure` tag as `username` (which is itself a word)